### PR TITLE
Eat your own dogfood

### PR DESCRIPTION
**Pull request by qznc**
Indented to tackle #176 but I do not expect this to be merged right away. However, we can have a discussion here, what maybe should be changed.

### CODE REVIEW

File: src/dfmt/formatter.d [hunk 1]
```diff
@@ -585,7 +591,8 @@ private:
             if (peekBackIs(tok!")"))
                 write(" ");
             auto e = expressionEndIndex(index);
-            immutable bool hasComment = tokens[index .. e].canFind!((a, b) => a.type == b)(tok!"comment");
+            immutable bool hasComment = tokens[index .. e].canFind!((a, b) => a.type == b)(
```
Comment (qznc) [inline src/dfmt/formatter.d]: Is this a sensible point to insert a line break?

File: src/dfmt/formatter.d [hunk 2]
```diff
@@ -471,15 +474,18 @@ private:
         import dfmt.editorconfig : OptionalBoolean;
 
         immutable bool isCase = astInformation.caseEndLocations.canFindIndex(current.index);
-        immutable bool isAttribute = astInformation.attributeDeclarationLines.canFindIndex(current.line);
+        immutable bool isAttribute = astInformation.attributeDeclarationLines.canFindIndex(
```
Comment (qznc) [inline src/dfmt/formatter.d]: Maybe a line break before ".canFindIndex" would be better?

File: src/dfmt/formatter.d [hunk 3]
```diff
@@ -1119,7 +1143,8 @@ private:
                 }
                 immutable l = indents.indentToMostRecent(tok!"switch");
                 if (l != -1)
-                    indentLevel = config.dfmt_align_switch_statements == OptionalBoolean.t ? l : indents.indentLevel;
+                    indentLevel = config.dfmt_align_switch_statements == OptionalBoolean.t ? l
```
Comment (qznc) [inline src/dfmt/formatter.d]: A line break before OptionalBoolean would maybe be better?

File: src/dfmt/formatter.d [hunk 4]
```diff
@@ -1172,8 +1197,8 @@ private:
             }
             else
             {
-                while (indents.topIsTemp() && (peekBackIsOneOf(true, tok!"}", tok!";")
-                        && indents.top != tok!";"))
+                while (indents.topIsTemp() && (peekBackIsOneOf(true, tok!"}",
```
Comment (qznc) [inline src/dfmt/formatter.d]: Might be a bug?

File: src/dfmt/tokens.d [hunk 5]
```diff
@@ -197,20 +196,21 @@ private string generateFixedLengthCases()
         "break", "byte", "case", "cast", "catch", "cdouble", "cent", "cfloat",
         "char", "class", "const", "continue", "creal", "dchar", "debug",
         "default", "delegate", "delete", "deprecated", "do", "double", "else",
```
Comment (qznc) [inline src/dfmt/tokens.d]: Should we use //dfmt off here?

File: src/dfmt/formatter.d (hunk 2 above)
Comment (Hackerpilot) [inline src/dfmt/formatter.d]: The algorithm is designed to prefer breaking lines at `(` over `.`. I don't think this is a bug.

File: src/dfmt/tokens.d (hunk 5 above)
Comment (Hackerpilot) [inline src/dfmt/tokens.d]: I don't see how this new formatting is much worse than the old one.

File: src/dfmt/formatter.d (hunk 2 above)
Comment (qznc) [inline src/dfmt/formatter.d]: Split off into #181  I have a commit, which splits at "=", but it breaks other tests at the moment.

**Comment by Hackerpilot** [comment]
I'm going to close this because:
* It got very out-of-date
* I have run dfmt on itself recently

**Pull request closed without merging**